oracle.go

```diff
@@ -280,6 +280,72 @@ func (o *Oracle) RotateRootCredentials(ctx context.Context, statements []string)
 	return o.RawConfig, nil
 }
 
+func (o *Oracle) SetCredentials(ctx context.Context, statements dbplugin.Statements, staticUser dbplugin.StaticUserConfig) (username, password string, err error) {
+	rotateStatements := statements.Rotation
+	if len(rotateStatements) == 0 {
+		rotateStatements = []string{`ALTER USER "{{username}}" IDENTIFIED BY "{{password}}"`}
+	}
+
+	username = staticUser.Username
+	password = staticUser.Password
+	if username == "" || password == "" {
+		return "", "", errors.New("must provide both username and password")
+	}
+
+	variables := map[string]string{
+		"username": strings.ToUpper(username),
```

Comment:
Why is `ToUpper` used?

Reply:
Oracle uppercases the username when you create it, but not when you log in with it. It's super intuitive.

I'll put a comment in describing why the `ToUpper` is here.

Reply:
Slight correction: Oracle uppercases the username when it's created. When you call something like `ALTER USER foo ...` it will auto-upper case for you. But if you do `ALTER USER "foo" ...` it does not. Additionally, when the user was lowercased, it was failing to connect to Oracle in the test container. Since the username is uppercased by Oracle, I decided to upper case it here for consistency.

I've also updated a couple of other places in the plugin so it uppercases the username before applying the query.

Reply:
How will this work for backwards compatibility? If my creation statement is `CREATE USER "{{username}}"...`, which would have been preserving case, and there currently exists a dynamics cred with username `"foo"`, after I upgrade won't the revoke now be passing in `"FOO"` and it won't match/revoke?

Reply:
After lengthy discussion with @kalafut offline, we decided the following:
1) Remove the explicit `ToUpper` calls to allow for mixed/lower case usernames if the user needs them
2) Remove the double quotes around the username in `ALTER USER "{{username}}" ...`
3) Add double quotes around the password to allow special characters that otherwise screw with Oracle
4) Use the same default query for `RotateRootCredentials` and `SetCredentials` for consistency
5) Add documentation explaining the case sensitivity problem in Oracle (both in the README and the Vault docs)